src/row.jl

```diff
@@ -131,3 +136,54 @@ Lighthouse <v0.14.0).
 function _evaluation_row_dict(row::EvaluationRow)
     return Dict(string(k) => v for (k, v) in pairs(NamedTuple(row)) if !ismissing(v))
 end
+
+#####
+##### `ObservationRow`
+#####
+
+# Redefinition is workaround for https://github.com/beacon-biosignals/Legolas.jl/issues/9
+const OBSERVATION_ROW_SCHEMA = Legolas.Schema("lighthouse.observation@1")
+"""
+    const ObservationRow = Legolas.@row("lighthouse.observation@1",
+                                        predicted_hard_label::Int64,
+                                        predicted_soft_labels::Vector{Float32},
+                                        elected_hard_label::Int64,
+                                        votes::Union{Missing,Vector{Int64}})
+
+A type alias for [`Legolas.Row{typeof(Legolas.Schema("lighthouse.observation@1"))}`](https://beacon-biosignals.github.io/Legolas.jl/stable/#Legolas.@row)
+representing the per-observation input values required to compute [`evaluation_metrics_row`](@ref).
+"""
+const ObservationRow = Legolas.@row("lighthouse.observation@1",
+                                    predicted_hard_label::Int64,
+                                    predicted_soft_labels::Vector{Float32},
+                                    elected_hard_label::Int64,
+                                    votes::Union{Missing,Vector{Int64}})
+
+function _observation_table_to_inputs(observation_table)
+    Legolas.validate(observation_table, OBSERVATION_ROW_SCHEMA)
+    df_table = DataFrame(observation_table; copycols=false)
+    votes = missing
+    if any(ismissing, df_table.votes) && !all(ismissing, df_table.votes)
+        throw(ArgumentError("`:votes` must either be all `missing` or contain no `missing`"))
+    end
+    votes = any(ismissing, df_table.votes) ? missing :
+            transpose(reduce(hcat, df_table.votes))
+
+    predicted_soft_labels = transpose(reduce(hcat, df_table.predicted_soft_label))
+    return (; predicted_hard_labels=df_table.predicted_hard_label, predicted_soft_labels,
+            elected_hard_labels=df_table.elected_hard_label, votes)
+end
+
+function _inputs_to_observation_table(; predicted_hard_labels::AbstractVector,
+                                      predicted_soft_labels::AbstractMatrix,
+                                      elected_hard_labels::AbstractVector,
+                                      votes::Union{Nothing,Missing,AbstractMatrix}=nothing)
+    votes = has_value(votes) ? collect(eachrow(votes)) : missing
+    observations = DataFrame(; predicted_hard_label=predicted_hard_labels,
+                             elected_hard_label=elected_hard_labels,
+                             predicted_soft_labels=collect(eachrow(predicted_soft_labels)),
+                             votes)
+    observation_table = DataFrame([ObservationRow(r) for r in eachrow(observations)])
```

Comment:
This is probably how I'd write this, to not need the DataFrames dependency:
```suggestion
    votes_itr = has_value(votes) ? eachrow(votes) : (missing for _ in 1:length(predicted_hard_labels))
    predicted_soft_labels_itr = eachrow(predicted_soft_labels_itr)
    if !(length(predicted_hard_labels) == length(predicted_soft_labels_itr) == length(elected_hard_labels) == length(votes_itr))
        throw(DimensionMismatch("Inputs do not all have the same number of observations"))
    end
    observation_table = map(predicted_hard_labels, elected_hard_labels, predicted_soft_labels_itr, votes_itr) do predicted_hard_label, elected_hard_label, predicted_soft_label, votes
        ObservationRow(; predicted_hard_label, elected_hard_label, predicted_soft_label, votes)
    end
```
Now we return a `Vector` of `ObservationRow`s, which is still a valid table. Then if the caller wants a column table, they can just call `Tables.columns` on it.

Reply:
oh, awesome. i'm going to leave DataFrames in the tests, I think, but will move it out of the main package!

Reply:
Ok, sounds good!